### PR TITLE
ELPP-2893 Generate critical CSS

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: salt/journal/critical-css.sls [hunk 1]
```diff
@@ -49,17 +49,26 @@ journal-local-demo-parameters:
         - require:
             - journal-local-demo-separate-folder
 
+journal-local-demo-critical-css:
```
Comment (thewilkybarkid) [inline salt/journal/critical-css.sls]: Temporary, a better solution is to make the folder exist in the repo with a `.gitkeep`. (Twig fails if it doesn't exist.)

File: salt/journal/critical-css.sls [hunk 2]
```diff
@@ -49,17 +49,26 @@ journal-local-demo-parameters:
...
         - cwd: /srv/journal-local-demo
         - user: {{ pillar.elife.deploy_user.username }}
         - require:
+            - journal-local-demo-critical-css
             - journal-local-demo-parameters
 
 generate-critical-css:
     cmd.run:
-        - name: echo "We will use gulp to generate critical CSS here"
+        - name: retry "node_modules/.bin/gulp critical-css:generate" 3
```
Comment (giorgiosironi) [inline salt/journal/critical-css.sls]: does this also need to retry? I thought it was the image generation that was failing

File: salt/journal/critical-css.sls (hunk 1 above)
Comment (giorgiosironi) [inline salt/journal/critical-css.sls]: `build/` content is ignored but files can be force-committed to it, no problem

Comment (giorgiosironi) [inline salt/journal/critical-css.sls]: given the duplication between the two /srv folders, best to commit it

**Pull request merged**